Index: dao/src/main/java/com/baifendian/swordfish/dao/mysql/mapper/MasterServerMapper.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- dao/src/main/java/com/baifendian/swordfish/dao/mysql/mapper/MasterServerMapper.java	(revision fc93c0e815a77e2ec4e12ec86ed5f00407740083)
+++ dao/src/main/java/com/baifendian/swordfish/dao/mysql/mapper/MasterServerMapper.java	(revision )
@@ -23,7 +23,7 @@
 @MapperScan
 public interface MasterServerMapper {
     @Results(value = {
-            @Result(property = "ip", column = "ip",javaType = String.class,jdbcType = JdbcType.VARCHAR),
+            @Result(property = "host", column = "host",javaType = String.class,jdbcType = JdbcType.VARCHAR),
             @Result(property = "port", column = "port",javaType = int.class,jdbcType = JdbcType.INTEGER),
             @Result(property = "updateTime", column = "update_time",javaType = int.class,jdbcType = JdbcType.INTEGER),
     })
@@ -32,7 +32,7 @@
 
 
     @InsertProvider(type = MasterServerMapperSQL.class, method = "insert")
-    int insert(@Param("session") Session session);
+    int insert(@Param("masterServer") MasterServer masterServer);
 
     @SelectProvider(type = MasterServerMapperSQL.class, method = "delete")
     void delete();
Index: dao/src/main/java/com/baifendian/swordfish/dao/MasterDao.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- dao/src/main/java/com/baifendian/swordfish/dao/MasterDao.java	(revision fc93c0e815a77e2ec4e12ec86ed5f00407740083)
+++ dao/src/main/java/com/baifendian/swordfish/dao/MasterDao.java	(revision )
@@ -8,6 +8,7 @@
 
 package com.baifendian.swordfish.dao;
 
+import com.baifendian.swordfish.common.utils.BFDDateUtils;
 import com.baifendian.swordfish.dao.mysql.MyBatisSqlSessionFactoryUtil;
 import com.baifendian.swordfish.dao.mysql.mapper.MasterServerMapper;
 import com.baifendian.swordfish.dao.mysql.model.MasterServer;
@@ -32,7 +33,11 @@
         return masterServerMapper.query();
     }
 
-    public registerMasterServer(String host, int port){
-        return masterServerMapper.insert()
+    public int registerMasterServer(String host, int port){
+        MasterServer masterServer = new MasterServer();
+        masterServer.setHost(host);
+        masterServer.setPort(port);
+        masterServer.setUpdateTime(BFDDateUtils.getSecs());
+        return masterServerMapper.insert(masterServer);
     }
 }
Index: web-server/src/main/java/com/baifendian/swordfish/webserver/WebThriftServer.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- web-server/src/main/java/com/baifendian/swordfish/webserver/WebThriftServer.java	(revision fc93c0e815a77e2ec4e12ec86ed5f00407740083)
+++ web-server/src/main/java/com/baifendian/swordfish/webserver/WebThriftServer.java	(revision )
@@ -46,12 +46,15 @@
 
     private TServer server;
 
-    private final String hostName;
+    private final String host;
 
     private final int port;
 
     private final String MASTER_PORT = "master.port";
 
+    private final String MASTER_MIN_THREADS = "master.min.threads";
+    private final String MASTER_MAX_THREADS = "master.max.threads";
+
     private MasterDao masterDao;
 
     static{
@@ -63,7 +66,7 @@
     }
 
     public WebThriftServer() throws UnknownHostException, TTransportException {
-        hostName = InetAddress.getLocalHost().getHostName();
+        host = InetAddress.getLocalHost().getHostName();
         port = conf.getInt(MASTER_PORT);
         masterDao = DaoFactory.getDaoInstance(MasterDao.class);
         init();
@@ -75,16 +78,18 @@
     private void init() throws UnknownHostException, TTransportException {
         TProtocolFactory protocolFactory = new TBinaryProtocol.Factory();
         TTransportFactory tTransportFactory = new TTransportFactory();
-        TProcessor tProcessor = new WorkerService.Processor(new MasterServiceImpl());
-        InetSocketAddress inetSocketAddress = new InetSocketAddress(hostName, port);
-        server = getTThreadPoolServer(protocolFactory, tProcessor, tTransportFactory, inetSocketAddress, 50, 200);
+        TProcessor tProcessor = new MasterService.Processor(new MasterServiceImpl());
+        InetSocketAddress inetSocketAddress = new InetSocketAddress(host, port);
+        int minThreads = conf.getInt(MASTER_MIN_THREADS, 50);
+        int maxThreads = conf.getInt(MASTER_MAX_THREADS, 200);
+        server = getTThreadPoolServer(protocolFactory, tProcessor, tTransportFactory, inetSocketAddress, minThreads, maxThreads);
         logger.info("start thrift server on port:{}", port);
 
         registerMaster();
     }
 
     public void registerMaster(){
-        masterDao.
+        masterDao.registerMasterServer(host, port);
     }
 
     public static void main(String[] args) {
Index: dao/src/main/java/com/baifendian/swordfish/dao/mysql/model/MasterServer.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- dao/src/main/java/com/baifendian/swordfish/dao/mysql/model/MasterServer.java	(revision fc93c0e815a77e2ec4e12ec86ed5f00407740083)
+++ dao/src/main/java/com/baifendian/swordfish/dao/mysql/model/MasterServer.java	(revision )
@@ -14,18 +14,18 @@
  */
 public class MasterServer {
 
-    private String ip;
+    private String host;
 
     private int port;
 
     private int updateTime;
 
-    public String getIp() {
-        return ip;
+    public String getHost() {
+        return host;
     }
 
-    public void setIp(String ip) {
-        this.ip = ip;
+    public void setHost(String host) {
+        this.host = host;
     }
 
     public int getPort() {
